pkg/render/logstorage.go

```diff
@@ -1705,6 +1773,184 @@ func (es elasticsearchComponent) oidcUserRoleBinding() []client.Object {
 	}
 }
 
+func (es elasticsearchComponent) esGatewayDeployment() *appsv1.Deployment {
```

Comment:
I don't think this should be added to the this component, I think it should be it's own component in the logstorage package. You can render multiple components per controller, this component is bloated and we should add anything to it that is elasticsearch specific (i.e. that actually configures Elasticsearch).